### PR TITLE
Chore: Refactor backend plugin manager/tsdb query data

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
Started a POC of some other things, but had the need move QueryData into backend plugin manager.
- This comes with the addition of tsdb.Service.HandleRequest becoming easier to reason about now since all data source queries hits this method and it will either call "legacy core backend plugins" or fallback to the backend plugin manager if there's no "legacy core backend plugin" registered/matching the requested plugin id.
- 2 wrappers/adapters for translating between legacy tsdb contracts became one and now lives in the tsdb package
- Removed and cleaned up this awkward need of calling GetDataPlugin in multiple places.

**Which issue(s) this PR fixes**:
Ref #21510

**Special notes for your reviewer**:
I do believe that with these changes we can now begin our journey of making the SDK contracts/data frames the optimal path if calling /api/ds/query. Basically replacing this as a start with a method optimal for handling and returning SDK request:
- https://github.com/grafana/grafana/blob/dfc466052d9c6fafbebaaf6691bab0caa1287e17/pkg/api/metrics.go#L77

### CODE REVIEW

File: pkg/plugins/datasource_plugin_wrapper_v2.go [hunk 1]
```diff
@@ -1,99 +0,0 @@
...
-	instanceSettings, err := adapters.ModelToInstanceSettings(ds)
-	if err != nil {
-		return DataResponse{}, err
-	}
-
-	if query.Headers == nil {
-		query.Headers = make(map[string]string)
-	}
-
-	if oauthtoken.IsOAuthPassThruEnabled(ds) {
```
Comment (wbrowne) [inline pkg/plugins/datasource_plugin_wrapper_v2.go:38]: We wanna keep this stuff right? Or have I missed where this is kept in the diff? 🤔

File: pkg/tsdb/service_test.go [hunk 2]
```diff
@@ -99,17 +114,24 @@ func (e *fakeExecutor) HandleQuery(refId string, fn resultsFn) {
 
 type fakeBackendPM struct {
 	backendplugin.Manager
+	backend.QueryDataHandlerFunc
 }
 
-func (pm fakeBackendPM) GetDataPlugin(string) interface{} {
-	return nil
+func (m *fakeBackendPM) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	fmt.Println("QueryData called")
```
Comment (wbrowne) [inline pkg/tsdb/service_test.go:121]: Nit: Can remove this?

File: pkg/tsdb/service_test.go [hunk 3]
```diff
@@ -48,16 +51,28 @@ func TestHandleRequest(t *testing.T) {
...
-		svc, _ := createService()
+	t.Run("Should fallback to backend plugin manager when handling request for query with unregistered type", func(t *testing.T) {
+		svc, _, manager := createService()
+		backendPluginManagerCalled := false
+		manager.QueryDataHandlerFunc = backend.QueryDataHandlerFunc(func(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+			backendPluginManagerCalled = true
+			return &backend.QueryDataResponse{}, nil
+		})
 
+		ds := &models.DataSource{Id: 12, Type: "testjughjgjg", JsonData: simplejson.New()}
```
Comment (wbrowne) [inline pkg/tsdb/service_test.go:62]: Nit: 
```suggestion
		ds := &models.DataSource{Id: 12, Type: "unregisteredType", JsonData: simplejson.New()}
```

File: pkg/plugins/datasource_plugin_wrapper_v2.go (hunk 1 above)
Comment (marefr) [reply pkg/plugins/datasource_plugin_wrapper_v2.go:38]: Good catch :pray: Fixed

File: pkg/plugins/backendplugin/manager/manager.go [hunk 4]
```diff
@@ -244,10 +244,7 @@ func (m *manager) stop(ctx context.Context) {
 
 // CollectMetrics collects metrics from a registered backend plugin.
 func (m *manager) CollectMetrics(ctx context.Context, pluginID string) (*backend.CollectMetricsResult, error) {
-	m.pluginsMu.RLock()
-	p, registered := m.plugins[pluginID]
-	m.pluginsMu.RUnlock()
-
+	p, registered := m.Get(pluginID)
```
Comment (wbrowne) [inline pkg/plugins/backendplugin/manager/manager.go:247]: Mental note that `Get` should also probably check `IsDecommissioned()` as part of the second return. Something for later

**Review by wbrowne — APPROVED** [review]
Great stuff! 

PS you can also remove `(m *manager) GetDataPlugin(pluginID string) interface{}`

File: pkg/plugins/backendplugin/manager/manager.go (hunk 4 above)
Comment (marefr) [reply pkg/plugins/backendplugin/manager/manager.go:235]: I pushed an update to `Get` returning nil, and false if the plugin is decommissioned

**Review by wbrowne — APPROVED** [review]
Tested once again and looks good :)

**Review by torkelo — COMMENTED** [review]
Nice!

**Review by sunker — APPROVED** [review]
Nice! Tested and it works fine.

**Pull request merged**